src/kafka/protocol/messages.js

```diff
@@ -202,6 +239,89 @@ const parseCommand = (messageProtocol) => {
   }
 }
 
+/**
+ * Encodes Payload to base64 encoded data URI
+ *
+ * @param {buffer\|string} input - Buffer or String
+ * @param {MimeTypes} mimeType - mime type of the input
+ *
+ * @return {string} - Returns base64 encoded data  URI string
+ */
+
+exports.encodePayload = (input, mimeType) => {
+  if (allowedMimeTypes.includes(mimeType)) {
+    return (input instanceof Buffer)
+      ? `data:${mimeType};base64,${base64url(input, 'utf8')}`
+      : `data:${mimeType};base64,${base64url(Buffer.from(input), 'utf8')}`
+  } else {
+    throw new Error(`mime type should be one of the following:${allowedMimeTypes.map(e => ` ${e}`)}`)
+  }
+}
+
+/**
+ * Decode Payload to base64 encoded data URI
+ *
+ * @param {string} input - JSON as Data URI or plain string
```

Comment:
This description is not accurate....string can be anything based on the dataURI mimetype.